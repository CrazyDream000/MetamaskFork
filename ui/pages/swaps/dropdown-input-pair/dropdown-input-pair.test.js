import React from 'react';
import configureMockStore from 'redux-mock-store';

import {
  renderWithProvider,
  createSwapsMockStore,
} from '../../../../test/jest';
import DropdownInputPair from '.';

const createProps = (customProps = {}) => {
  return {
    ...customProps,
  };
};

describe('DropdownInputPair', () => {
  it('renders the component with initial props', () => {
    const store = configureMockStore()(createSwapsMockStore());
    const props = createProps();
    const { getByPlaceholderText } = renderWithProvider(
      <DropdownInputPair {...props} />,
      store,
    );
    expect(getByPlaceholderText('0')).toBeInTheDocument();
    expect(
      document.querySelector('.dropdown-input-pair__input'),
    ).toMatchSnapshot();
  });
});
