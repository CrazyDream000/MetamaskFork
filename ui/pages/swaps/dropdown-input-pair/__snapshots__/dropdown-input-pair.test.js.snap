// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownInputPair renders the component with initial props 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root dropdown-input-pair__input MuiFormControl-marginDense MuiFormControl-fullWidth"
>
  <div
    class="MuiInputBase-root MuiInput-root TextField-inputRoot-12 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
      dir="auto"
      placeholder="0"
      type="text"
      value=""
    />
  </div>
</div>
`;
